dnf-docker-test/features/steps/updateinfo_utils.py

```diff
@@ -0,0 +1,178 @@
+    cnt.setAttribute("type", "self")
+    cnt.setAttribute("title", update['Id'])
+    references.appendChild(cnt)
+    if 'Reference' in update:
+        for ref in update['Reference']:
+            cnt = doc.createElement("reference")
+            cnt.setAttribute("href", "www.path.to/%s" % ref)
+            cnt.setAttribute("title", ref)
+            if ref.startswith('BZ'):
+                cnt.setAttribute("id", ref[2:])
+                cnt.setAttribute("type", 'bugzilla')
+            if ref.startswith('CVE'):
+                cnt.setAttribute("id", ref)
+                cnt.setAttribute("type", 'cve')
+            references.appendChild(cnt)
+    elem_update.appendChild(references)
+    # pkglist
+    pkglist = doc.createElement("pkglist")
+    elem_update.appendChild(pkglist)
+    # collection
+    collection = doc.createElement("collection")
+    collection.setAttribute("short", update.get('Collection', 'Default collection'))
+    pkglist.appendChild(collection)
+    # collection <name>
+    elem = doc.createElement("name")
+    cnt = doc.createTextNode(update.get('Collection', 'Default collection'))
+    elem.appendChild(cnt)
+    collection.appendChild(elem)
+    # now for every package I need to prepare this based on the details obtained from primary.xml
+    # <package name="PKG_NAME" version="3.0.33" release="3.29.el5_5.1" epoch="0" arch="i386" src="samba-3.0.33-3.29.el5_5.1.src.rpm">
+    # <filename>libsmbclient-3.0.33-3.29.el5_5.1.i386.rpm</filename>
+    # <sum type="md5">b5ea308c42fa07a4e59dd7b00c6a9db8</sum>
+    # </package>
+    for pkg in update['Package']:
+        # find the best match in available rpms stored in pkg_details dict
+        rpms = [rpm for rpm in pkg_details.keys() if rpm.startswith(pkg)]
+        rpms.sort()
+        rpm = rpms[-1]  # take the last one as that should be the _latest_ version
+        # create new package element
+        package = doc.createElement("package")
+        package.setAttribute("name", pkg_details[rpm]['pkg_name'])
+        package.setAttribute("version", pkg_details[rpm]['pkg_version'])
+        package.setAttribute("release", pkg_details[rpm]['pkg_release'])
+        package.setAttribute("epoch", pkg_details[rpm]['pkg_epoch'])
+        package.setAttribute("arch", pkg_details[rpm]['pkg_arch'])
+        package.setAttribute("src", pkg_details[rpm]['pkg_src'])
+        elem = doc.createElement("filename")
+        cnt = doc.createTextNode(rpm)
+        elem.appendChild(cnt)
+        package.appendChild(elem)
+        elem = doc.createElement("sum")
+        elem.setAttribute("type", pkg_details[rpm]['pkg_sumtype'])
+        cnt = doc.createTextNode(pkg_details[rpm]['pkg_sum'])
+        elem.appendChild(cnt)
+        package.appendChild(elem)
+        # add package into collection
+        collection.appendChild(package)
+    return elem_update
+
+
+def get_updateinfo_xml(repository, updateinfo_table):
+    primary_xml = get_repodata_file_path('.*primary.xml.gz$', repository)
```

Comment:
Well, I need to open a file I am able to uncompress. Currently there would be *.sqlite and *.gz files, we can add the support for other formats later but currently *.gz seems to be the best choice.